discovery-provider/src/models.py

```diff
@@ -169,31 +207,36 @@ def __repr__(self):
 class Track(Base):
     __tablename__ = "tracks"
 
+    # lazy init-ed by validates decorator the first time we need to do a model validation.
+    # used to map column to type since field is a string in decorator and __table__.columns
+    # returns a list
+    field_to_type_dict = {}
+
     blockhash = Column(String, ForeignKey("blocks.blockhash"), nullable=False)
     blocknumber = Column(Integer, ForeignKey("blocks.number"), nullable=False)
     track_id = Column(Integer, nullable=False)
     is_current = Column(Boolean, nullable=False)
     is_delete = Column(Boolean, nullable=False)
     owner_id = Column(Integer, nullable=False)
     route_id = Column(String, nullable=False)
-    title = Column(Text)
-    length = Column(Integer)
-    cover_art = Column(String)
-    cover_art_sizes = Column(String)
-    tags = Column(String)
-    genre = Column(String)
-    mood = Column(String)
-    credits_splits = Column(String)
+    title = Column(Text, nullable=True)
+    length = Column(Integer, nullable=True)
+    cover_art = Column(String, nullable=True)
+    cover_art_sizes = Column(String, nullable=True)
+    tags = Column(String, nullable=True)
+    genre = Column(String, nullable=True)
+    mood = Column(String, nullable=True)
+    credits_splits = Column(String, nullable=True)
     remix_of = Column(postgresql.JSONB, nullable=True)
-    create_date = Column(String)
-    release_date = Column(String)
-    file_type = Column(String)
-    description = Column(String)
-    license = Column(String)
-    isrc = Column(String)
-    iswc = Column(String)
+    create_date = Column(String, nullable=True)
+    release_date = Column(String, nullable=True)
+    file_type = Column(String, nullable=True)
+    description = Column(String, nullable=True)
+    license = Column(String, nullable=True)
+    isrc = Column(String, nullable=True)
+    iswc = Column(String, nullable=True)
     track_segments = Column(postgresql.JSONB, nullable=False)
-    metadata_multihash = Column(String)
+    metadata_multihash = Column(String, nullable=True)
     download = Column(postgresql.JSONB, nullable=True)
```

Comment:
if we're setting smarter defaults with jsonb type columns, should `nullable=False`?

Reply:
this is a db level change we need a migration for. i think schema check is abstracted from db check